File: libs/document-processor-next/src/__tests__/models/remove-property.ts
import { z } from 'zod';
import { withTypeBlueId } from '@blue-labs/language';

import { handlerContractBaseSchema } from '../../model/shared/index.js';

export const removePropertySchema = withTypeBlueId('RemoveProperty')(
  handlerContractBaseSchema.extend({
    propertyKey: z.string(),
  })
);

export type RemoveProperty = z.infer<typeof removePropertySchema>;

File: references/java-sdk/src/test/java/blue/language/processor/model/RemoveProperty.java
package blue.language.processor.model;

import blue.language.model.TypeBlueId;
import blue.language.processor.model.HandlerContract;

@TypeBlueId("RemoveProperty")
public class RemoveProperty extends HandlerContract {

    private String propertyKey;

    public String getPropertyKey() {
        return propertyKey;
    }

    public void setPropertyKey(String propertyKey) {
        this.propertyKey = propertyKey;
    }
}
